### ISSUE TITLE
Why does environment defaults to development and not production?

### ISSUE THREAD

**Issue by marcofugaro** [ai/browserslist#224]
This is my browserslist config:

```js
"browserslist": {
  "production": [
    "last 2 versions",
    "ie 11"
  ],
  "development": [
    "last 1 Chrome version"
  ]
}
```

My use-case is that I have a `staging` NODE_ENV which compiles my app as it was production, but doesn't uglify the code and uses sourcemaps. And of course I wanted to use the production browserlist.

Instinctively, I thought that if my NODE_ENV didn't match any of the two options, it would default to production, but instead it defaults to development.

https://github.com/ai/browserslist/blob/64e9c07b6186d5eb4cfdeb8b5e3dc9c2bc6c8b43/node.js#L61

What did bring you to that decision? I think it can be reconsidered since the development environment is always one and only, instead you can have multiple production environments, say `test` or whatever else, where you would need to use the production browserslist, since it isn't viewed only by your development browser.

What do you think?

**Comment by ai** [comment]
Hard choice :). Can I ask you to make investigate what is default mode in Webpack 4 and create react app? I think we should use the most expected mode.

**Comment by ai** [comment]
Also, you are lucky to send this issue right before 3.0 release. We can break API.

@gaearon, since you create an idea of environments and CRA, may be the biggest user of it, what do you think?

**Comment by marcofugaro** [comment]
Sure! In the configless webpack4 the default `mode` is production.

> Use production defaults when omiting the mode option

See it [here under Features > Defaults](https://github.com/webpack/webpack/releases)

Create-react-app uses those envs by default:
- `NODE_ENV=development` when running `yarn start`
- `NODE_ENV=production` when running `yarn build`
- `NODE_ENV=test` when running `yarn test`

**Comment by ai** [comment]
Awesome! Do you want to send PR to save your name in the project history? :)

**Comment by ai** [comment]
I will release 3.0 when Jest release my Node.js fix

**Comment by ai** [comment]
Released in 3.0.
